Tests/PubNubTests/Networking/Routers/PresenceRouterTests.swift

```diff
@@ -357,3 +377,133 @@ final class PresenceEndpointTests: XCTestCase {
+    let router = PresenceRouter(.leave(channels: [channelName], groups: []), configuration: config)
+
+    XCTAssertEqual(router.endpoint.channels, [channelName])
+  }
+
+  func testLeave_Router_Groups() {
+    let router = PresenceRouter(.leave(channels: [], groups: [channelName]), configuration: config)
+
+    XCTAssertEqual(router.endpoint.groups, [channelName])
+  }
+}
+
+
+// MARK: - Get State Tests
+
+extension PresenceRouterTests {
+  func testGetState_Router() {
+    let router = PresenceRouter(.getState(uuid: "TestUUID", channels: [channelName], groups: []), configuration: config)
+
+    XCTAssertEqual(router.endpoint.description, "Get Presence State")
+    XCTAssertEqual(router.category, "Get Presence State")
+    XCTAssertEqual(router.service, .presence)
+  }
+
+  func testGetState_Router_ValidationError() {
+    let router = PresenceRouter(.getState(uuid: "", channels: [channelName], groups: []), configuration: config)
+    XCTAssertEqual(router.validationError?.pubNubError?.details.first,
+                      ErrorDescription.emptyUUIDString)
+
+    let missingChannelsGroups = PresenceRouter(.getState(uuid: "TestUUID", channels: [], groups: []), configuration: config)
+    XCTAssertEqual(missingChannelsGroups.validationError?.pubNubError?.details.first,
+                   ErrorDescription.missingChannelsAnyGroups)
+  }
+
+  func testGetState_Router_Channels() {
+    let router = PresenceRouter(.getState(uuid: "TestUUID", channels: [channelName], groups: []), configuration: config)
+
+    XCTAssertEqual(router.endpoint.channels, [channelName])
+  }
+
+  func testGetState_Router_Groups() {
+    let router = PresenceRouter(.getState(uuid: "TestUUID", channels: [], groups: [channelName]), configuration: config)
+
+    XCTAssertEqual(router.endpoint.groups, [channelName])
+  }
+}
+
+
+// MARK: - Set State Tests
+
+extension PresenceRouterTests {
+  func testSetState_Router() {
+    let router = PresenceRouter(.setState(channels: [channelName], groups: [], state: [:]), configuration: config)
+
+    XCTAssertEqual(router.endpoint.description, "Set Presence State")
+    XCTAssertEqual(router.category, "Set Presence State")
+    XCTAssertEqual(router.service, .presence)
+  }
+
+  func testSetState_Router_ValidationError() {
+    let router = PresenceRouter(.setState(channels: [], groups: [], state: [:]), configuration: config)
+
+    XCTAssertEqual(router.validationError?.pubNubError?.details.first,
+                      ErrorDescription.missingChannelsAnyGroups)
+  }
+
+  func testSetState_Router_Channels() {
+    let router = PresenceRouter(.setState(channels: [channelName], groups: [], state: [:]), configuration: config)
+
+    XCTAssertEqual(router.endpoint.channels, [channelName])
+  }
+
+  func testSetState_Router_Groups() {
+    let router = PresenceRouter(.setState(channels: [], groups: [channelName], state: [:]), configuration: config)
+
+    XCTAssertEqual(router.endpoint.groups, [channelName])
+  }
+}
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [File should contain 400 lines or less: currently contains 509](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4513555)